### ISSUE TITLE
Nightly releases are published even if E2E tests fail 

### ISSUE THREAD

**Issue by adrcunha** [knative/build-pipeline#361]
e.g. https://storage.googleapis.com/knative-prow/logs/ci-knative-build-pipeline-nightly-release/1080018717502345217/build-log.txt

E2E tests are not propagating the return code correctly. In the log above, the sequence is:

```
I0101 08:58:21.606] FAIL	github.com/knative/build-pipeline/test	1107.544s
I0101 08:58:21.618] Finished run, return code is 1
...
I0101 08:58:47.592] =============================================
I0101 08:58:47.592] ==== RUNNING YAML E2E TESTS FOR TASKRUNS ====
I0101 08:58:47.593] =============================================
...
I0101 08:58:53.679] **************************************
I0101 08:58:53.679] ***        E2E TESTS PASSED        ***
I0101 08:58:53.679] **************************************
I0101 08:58:53.681] =======================================
I0101 08:58:53.682] ==== TEARING DOWN TEST ENVIRONMENT ====
I0101 08:58:53.682] =======================================
I0101 08:58:53.824] ***************************************
I0101 08:58:53.824] ***         E2E TEST FAILED         ***
I0101 08:58:53.824] ***    Start of information dump    ***
I0101 08:58:53.825] ***************************************
I0101 08:58:53.825] >>> All resources:
...
I0101 09:04:21.184] Test subprocess exited with code 1
I0101 09:04:22.923] Test result code is 0
```

go tests failed, YAML tests passed. The teardown correcty shows that tests failed, but the test result is 0. Looks like the YAML tests overrode the test result.

**Comment by bobcatfish** [comment]
I think that this was (is?) resulting in pre-merge checks not catching end to end test failures as well (#368) so we really need to get this fixed asap @adrcunha  (unless it's fixed already!)

**Comment by adrcunha** [comment]
@bobcatfish  #370 will fix this and #368, please review at your earliest convenience.
